File: username and password/big_passwords_dict/generate_dict.go
package main

import (
	"bufio"
	"flag"
	"fmt"
	"os"
	"strings"
)

func main() {
	inputFilename := flag.String("i", "big_password_dict.txt", "input dictionary file")
	outputFilename := flag.String("o", "big_password_dict_generate.txt", "output dictionary file")
	listFilename := flag.String("l", "", "username list file")
	username := flag.String("u", "", "username")
	flag.Parse()

	passwordsInput := make(map[string]bool)
	inputF, err := os.Open(*inputFilename)
	if err != nil {
		fmt.Fprintln(os.Stderr, "error opening input file:", err)
		os.Exit(1)
	}
	defer inputF.Close()
	inputScanner := bufio.NewScanner(inputF)
	for inputScanner.Scan() {
		passwordsInput[strings.TrimSpace(inputScanner.Text())] = true
	}

	usernames := make(map[string]bool)
	if *listFilename != "" {
		listF, err := os.Open(*listFilename)
		if err != nil {
			fmt.Fprintln(os.Stderr, "error opening username list file:", err)
			os.Exit(1)
		}
		defer listF.Close()
		listScanner := bufio.NewScanner(listF)
		for listScanner.Scan() {
			usernames[strings.TrimSpace(listScanner.Text())] = true
		}
	} else {
		usernames[*username] = true
	}
	// } else if *username != "" {
	//     usernames[*username] = true
	// } else {
	//     fmt.Fprintln(os.Stderr, "error: either --list or --user must be specified")
	//     os.Exit(1)
	// }

	passwordOutput := make(map[string]bool)
	n := 0
	for username := range usernames {
		for password := range passwordsInput {
			if strings.Contains(password, "%user") {
				password = strings.ReplaceAll(password, "%user", username)
			}
			if !passwordOutput[password] {
				passwordOutput[password] = true
			}
			n++
			fmt.Printf("%d\r", n)
		}
	}

	outputF, err := os.Create(*outputFilename)
	if err != nil {
		fmt.Fprintln(os.Stderr, "error creating output file:", err)
		os.Exit(1)
	}
	defer outputF.Close()
	outputWriter := bufio.NewWriter(outputF)
	for password := range passwordOutput {
		fmt.Fprintln(outputWriter, password)
	}
	outputWriter.Flush()
}

File: username and password/big_passwords_dict/generate_dict.py
import argparse
import os

parser = argparse.ArgumentParser()
parser.add_argument("--list", "-l", help="Username list file", required=False)
parser.add_argument("--user", "-u", help="Username", required=False, default="")
parser.add_argument("--input", "-i", help="Input dictionary file", required=False, default="big_password_dict.txt")
parser.add_argument("--output", "-o", help="Output dictionary file", required=False, default="big_password_dict_generate.txt")
args = parser.parse_args()

with open(args.input, encoding="utf-8") as f:
    passwords_input = {i.strip() for i in f}

if args.list:
    with open(args.list, encoding="utf-8") as f:
        usernames = {i.strip() for i in f}
else:
    usernames = {args.user}

password_output = set()
n = 0
for i in usernames:
    for j in passwords_input:
        if "%user" in j:
            j = j.replace("%user", i)
        if j not in password_output:
            password_output.add(j)
        n += 1
        print(f"{n}\r", end="")

with open(args.output, "w", encoding="utf-8") as f:
    f.write("\n".join(password_output))
